pkg/apis/flowcontrol/types.go

```diff
@@ -88,8 +93,8 @@ type FlowSchemaSpec struct {
 	PriorityLevelConfiguration PriorityLevelConfigurationReference
 	// `matchingPrecedence` is used to choose among the FlowSchemas that match a given request. The chosen
 	// FlowSchema is among those with the numerically lowest (which we take to be logically highest)
-	// MatchingPrecedence.  Each MatchingPrecedence value must be non-negative.
-	// Note that if the precedence is not specified or zero, it will be set to 1000 as default.
+	// MatchingPrecedence.  Each MatchingPrecedence value must be ranged in [1,10000].
+	// Note that if the precedence is not specified, it will be set to 1000 as default.
```

Comment:
This feels like way too much information for a potential configuration author to keep in their head.

Do we really need to make the precedence optional? If so, why is the default 10% of the way through the possible range? Is anyone reading a config file ever going to remember that piece of trivia?

Do we really need exactly five digits of precision in precedences? (Why not either something far more constrained, like 1-100, or the full 32 bits?) 

If we require the precedence to be non-negative, why is it signed?

Reply:
The default shouldn't be at the logical top of the precedence range, other than that it doesn't matter too much.

I agree that it's not necessary to default this and we could make people choose something. OTOH this only matters if your flow schemas are overlapping.

We don't use unsigned variables in the API with few exceptions (port numbers), cargo-culting the go preference for signed ints to minimize under/overflow bugs.

Reply:
10k levels is probably way more than we need-- it was chosen due to being more readable than 2^31. In the interest of making progress, IMO 10k is good enough for alpha. It's better to have more than needed than less.

Reply:
Yes, in my opinion having a default value for matchingPrecedence just adds an unnecessary detail to the API.  Since it is quite plausible that some flow schemas overlap in their matching condition, everybody working on them has to be cognizant of the fact that they are ordered.  Remembering which number corresponds to unspecified matchingPrecedence is just additional burden.

Reply:
Regarding the range of matchingPrecedence: my preference is to not try to figure out how small we can make it.  We have not even begun to explore what configs admins will actually find useful, and that includes not yet exploring how they manage the matchingPrecedence space.  My preference is to leave admins as much freedom as `int32` allows.

Reply:
If we have a default value for matchingPrecedence then I think it should: (a) roughly correspond to ordinary workload in the suggested config and (b) have plenty of room above and below.